app/src/main/java/com/gh4a/activities/CommitHistoryActivity.java

```diff
@@ -52,14 +57,47 @@ protected void onInitExtras(Bundle extras) {
 
     @Override
     protected Fragment onCreateFragment() {
-        return CommitListFragment.newInstance(mRepoOwner, mRepoName, mRef, mFilePath);
+        CommitListFragment f = CommitListFragment.newInstance(mRepoOwner, mRepoName, mRef, mFilePath);
+        f.setFollowFileRenames(mFollowRenames);
+        return f;
     }
 
     @Override
     protected Intent navigateUp() {
         return RepositoryActivity.makeIntent(this, mRepoOwner, mRepoName, mRef);
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.commit_history_menu, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onPrepareOptionsMenu(Menu menu) {
+        MenuItem followItem = menu.findItem(R.id.follow_renames);
+        if (followItem != null) {
+            followItem.setChecked(mFollowRenames);
+        }
+        return super.onPrepareOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.follow_renames) {
+            mFollowRenames = !mFollowRenames;
+            item.setChecked(mFollowRenames);
```

Comment:
Question: store the setting persistently (similarly to 'wrap lines' in file viewer) or keep default as 'off'?

Reply:
I think it's fine as-is, see my comment below